test/nightly/crud.test.ts

```diff
@@ -0,0 +1,127 @@
+import { AzExtParentTreeItem, createResourceClient, createResourceGroup, deleteResourceGroupV2, ext, IActionContext, LocationListStep, randomUtils, settingUtils, SubscriptionItem } from '../../extension.bundle';
+import { longRunningTestsEnabled } from "../global.test";
+import assert = require("assert");
+
+let rgName: string;
+let locations: Location[];
+let testSubscription: SubscriptionItem;
+
+suite('Resource CRUD Operations', function (this: Mocha.Suite): void {
+    this.timeout(7 * 60 * 1000);
+
+    suiteSetup(async function (this: Mocha.Context): Promise<void> {
+        if (!longRunningTestsEnabled) {
+            this.skip();
+        }
+
+        ext.testing.overrideAzureServiceFactory = undefined;
+        ext.testing.overrideAzureSubscriptionProvider = undefined;
+
+        const subscriptionTreeItems = await ext.appResourceTree.getChildren() as unknown as SubscriptionItem[];
+        if (subscriptionTreeItems.length > 0) {
+            const testContext = await createTestActionContext();
+            testSubscription = subscriptionTreeItems[0] as SubscriptionItem;
+            const context = { ...testContext, ...testSubscription.subscription };
+            locations = await LocationListStep.getLocations(context);
+        }
+
+        rgName = randomUtils.getRandomHexString(12);
+    });
+
+    test('Create Resource Group (single)', async () => {
+        await runWithTestActionContext('createResourceGroup', async context => {
+            const testInputs: (string | RegExp)[] = [rgName, locations[0].displayName!];
+            await context.ui.runWithInputs(testInputs, async () => {
+                await createResourceGroup(context, testSubscription);
+            });
+
+            assert.ok(await resourceGroupExists(context, rgName));
+        });
+    });
+
+    test('Create Resource Groups (all locations)', async () => {
+        await Promise.all(locations.map(async l => {
+            await runWithTestActionContext('createResourceGroup', async context => {
+                const testInputs: (string | RegExp)[] = [`${rgName}-${l.name}`, l.displayName!];
+                await context.ui.runWithInputs(testInputs, async () => {
+                    await createResourceGroup(context, testSubscription);
+                });
+
+                assert.ok(await resourceGroupExists(context, `${rgName}-${l.name}`));
+            });
+        }));
+    });
+
+    test('Get Resources', async () => {
+        const subscriptionTreeItems = await ext.appResourceTree.getChildren();
+        assert.ok(subscriptionTreeItems.length > 0);
+        for (const subscription of subscriptionTreeItems) {
+            const groupTreeItems = await ext.appResourceTree.getChildren(subscription as AzExtParentTreeItem);
+            await Promise.all(groupTreeItems.map(async g => {
+                const children = await ext.appResourceTree.getChildren(g as AzExtParentTreeItem);
```

Comment:
Is it possible for an error to be thrown when calling `getChildren`?  Will it mess up the rest of the tests if an error is thrown here?